tests/proto/app/YBuild

```diff
@@ -63,10 +63,16 @@ Proto(
 
 ProtoCollector('hello1-collector', deps=':hello1-proto')
 
+GoPackage(
+    'hello-proto-lib',
+    sources=[],
+    deps=':hello-proto',
+    in_buildenv='//:proto-builder'
+)
+
 GoProg('hello-go',
        sources='hello.go',
-       deps=':hello-proto',
-       go_package='foo.com',
+       deps=':hello-proto-lib',
```

Comment:
Any reason not to put the deps directly on the Proto target (':hello-proto')?

Reply:
It depends on what we want. The GoPackage that depends on the proto is used to build the generated .go from the proto to check it compiles. If we don't want to check that the proto compiles, I can change this. In the current implementation, when there is a direct dependency in a proto target, it is assumed that it is a GoPackage that build the generated code from the proto.

Reply:
this means it adds the generated .go to the build command. In the current implementation, if a go target depends directly on a proto target, the build will fail.

Reply:
Maybe the correct thing is to trust the proto generator and skip building the generated code. It is rebuild anyway in each target.
Then all targets will depend directly in the proto without a middle package. Makes more sense.

Reply:
I removed building the generated .go from proto by itself. So now all go targets can depend directly in the proto.